mpcite/models.py

```diff
@@ -0,0 +1,378 @@
+        doi_collection_record.last_validated_on = datetime.now()
+        return doi_collection_record
+
+
+class ElsevierPOSTContainerModel(BaseModel):
+    identifier: str = Field(default="", title="mp_id")
+    source: str = Field(default="MATERIALS_PROJECT")
+    date: str = Field(default=datetime.now().date().isoformat().__str__())
+    title: str = Field(...)
+    description: str = Field(default="")
+    doi: str = Field(...)
+    authors: List[str] = Field(default=['Kristin Persson'])
+    url: str = Field(...)
+    type: str = Field(default='dataset')
+    dateAvailable: str = Field(default=datetime.now().date().isoformat().__str__())
+    dateCreated: str = Field(default=datetime.now().date().isoformat().__str__())
+    version: str = Field(default="1.0.0")
+    funding: str = Field(default='USDOE Office of Science (SC), Basic Energy Sciences (BES) (SC-22)')
+    language: str = Field(default="en")
+    method: str = Field(default="Materials Project")
+    accessRights: str = Field(default="Public")
+    contact: str = Field('Kristin Persson <kapersson@lbl.gov>')
+    dataStandard: str = Field(default="https://materialsproject.org/citing")
+    howToCite: str = Field(default="https://materialsproject.org/citing")
+    subjectAreas: List[str] = Field(default=["36 MATERIALS SCIENCE"])
+    keywords: List[str] = Field(...)
+    institutions: List[str] = Field(default=["Lawrence Berkeley National Laboratory"])
+    institutionIds: List[str] = Field(default=["AC02-05CH11231; EDCBEE"])
+    spatialCoverage: List[str] = Field(default=[])
+    temporalCoverage: List[str] = Field(default=[])
+    references: List[str] = Field(default=["https://materialsproject.org/citing"])
+    relatedResources: List[str] = Field(default=["https://materialsproject.org/citing"])
+    location: str = Field("1 Cyclotron Rd, Berkeley, CA 94720")
+    childContainerIds: List[str] = Field(default=[])
+
+    @classmethod
+    def get_url(cls, mp_id):
+        return 'https://materialsproject.org/materials/%s' % mp_id
+
+    @classmethod
+    def get_keywords(cls, material: MaterialModel):
+        if material.bandstructure is not None:
+            return ['crystal structure', material.formula_pretty, material.chemsys, 'electronic bandstructure']
+        else:
+            return ['crystal structure', material.formula_pretty, material.chemsys]
+
+    @classmethod
+    def get_default_description(cls):
+        return 'Computed materials data using density ' \
+               'functional theory calculations. These calculations determine ' \
+               'the electronic structure of bulk materials by solving ' \
+               'approximations to the Schrodinger equation. For more ' \
+               'information, see https://materialsproject.org/docs/calculations'
+
+    @classmethod
+    def get_date_created(cls, material: MaterialModel) -> str:
+        return material.created_at.date().__str__()
```

Comment:
This doesn't make sense with the above defaults. If you want a generator method that turns a `MaterialModel` into a pre-filled out `ElsevierPOSTContainerModel`, then it should be defined as such.

Reply:
you are right, legacy code that happened to work, I'll clean it up, sry for the mess